glommio/src/executor/mod.rs

```diff
@@ -592,10 +592,11 @@ impl LocalExecutorBuilder {
     /// [`LocalExecutor::run`]:struct.LocalExecutor.html#method.run
     #[must_use = "This spawns an executor on a thread, so you may need to call \
                   `JoinHandle::join()` to keep the main thread alive"]
-    pub fn spawn<G, F, T>(self, fut_gen: G) -> Result<JoinHandle<()>>
+    pub fn spawn<G, F, T>(self, fut_gen: G) -> io::Result<JoinHandle<io::Result<T>>>
```

Comment:
Shouldn't these Results be glommio::Error variants to explain to the user what happened? Also shouldn't we just give the user the ability to return a result of their choosing from the main future?

Reply:
We do! The `T` in there can be anything the user wants. They can return a `Result` if they so wish. The two errors you see are respectively:
* The error thrown by `join()` if the `std::thread::Thread` failed to create or panicked
* The error thrown by `LocalExecutor::new` if we fail to create it

Reply:
It still seems like we need to return a glommio Error type (not an io::Error, from the io::Result) for functions that are public like this.